### PR TITLE
emoji skin tone picker and other tweaks in emoji picker

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

File: shared/chat/conversation/messages/react-button/emoji-picker/container.tsx [hunk 1]
```diff
@@ -52,100 +53,147 @@ const useReacji = ({onPick, onDidPick}: Props) => {
...
+      <Kb.Box2 direction="horizontal" fullWidth={true} alignItems="center">
+        <Kb.ClickableBox onClick={onCancel} style={styles.cancelContainerMobile}>
+          <Kb.Text type="BodyBigLink">Cancel</Kb.Text>
+        </Kb.ClickableBox>
+        <Kb.SearchFilter
+          focusOnMount={true}
+          size="small"
+          icon="iconfont-search"
+          placeholderText="Search"
+          onChange={str => setFilter(str)}
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/react-button/emoji-picker/container.tsx]: can likely just be `setFilter`

File: shared/chat/conversation/messages/react-button/emoji-picker/container.tsx [hunk 2]
```diff
@@ -52,100 +53,147 @@ const useReacji = ({onPick, onDidPick}: Props) => {
...
+          style={styles.searchFilter}
+        />
+      </Kb.Box2>
       <EmojiPicker
         topReacjis={topReacjis}
         filter={filter}
-        onChoose={emoji => onAddReaction(`:${emoji.short_name}:`)}
+        onChoose={emoji => onAddReaction(emoji)}
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/react-button/emoji-picker/container.tsx]: can likely be `onAddReaction`

File: shared/chat/conversation/messages/react-button/emoji-picker/container.tsx [hunk 3]
```diff
@@ -52,100 +53,147 @@ const useReacji = ({onPick, onDidPick}: Props) => {
...
+        />
+        <SkinTonePicker currentSkinTone={currentSkinTone} setSkinTone={setSkinTone} />
+      </Kb.Box2>
       <Kb.Box style={styles.emojiContainer}>
         <EmojiPicker
           topReacjis={topReacjis}
           filter={filter}
-          onChoose={emoji => onAddReaction(`:${emoji.short_name}:`)}
-          width={320}
+          onChoose={emoji => onAddReaction(emoji)}
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/react-button/emoji-picker/container.tsx]: same

File: shared/chat/conversation/messages/react-button/emoji-picker/container.tsx [hunk 4]
```diff
@@ -52,100 +53,147 @@ const useReacji = ({onPick, onDidPick}: Props) => {
...
+          width={336}
+          skinTone={currentSkinTone}
         />
       </Kb.Box>
+      {/* TODO
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/react-button/emoji-picker/container.tsx]: ?

File: shared/constants/types/chat2/index.tsx [hunk 5]
```diff
@@ -177,6 +177,10 @@ export type ParticipantInfo = {
   contactName: Map<string, string> // member username -> contact name
 }
 
+// Corresponds to skinTones in emoji-datasource.
+export type EmojiSkinToneNonDefault = '1F3FB' | '1F3FC' | '1F3FD' | '1F3FE' | '1F3FF'
+export type EmojiSkinTone = 'default' | EmojiSkinToneNonDefault
```
Comment (chrisnojima) [inline shared/constants/types/chat2/index.tsx]: why not just use undefined and have it be optional in the component

File: shared/chat/conversation/messages/react-button/emoji-picker/container.tsx (hunk 4 above)
Comment (songgao) [reply shared/chat/conversation/messages/react-button/emoji-picker/container.tsx]: This is the footer on desktop. I'll do that in a different ticket.

**Pull request merged**